tests/unit/channel_test.py

```diff
@@ -68,3 +72,82 @@ def test_read_bulk_response(ssh_client_session):
             break
 
     assert b'Hello World' in response  # noqa: WPS302
+
+
+def test_request_exec(ssh_channel):
+    """Test direct call to request_exec."""
+    ssh_channel.request_exec('exit 1')
+
+    rc = -1
+    while rc == -1:
+        ssh_channel.poll(timeout=5)
+        rc = ssh_channel.get_channel_exit_status()
+    assert rc == 1
+
+
+def test_send_eof(ssh_channel):
+    """Test send_eof correctly terminates input stream."""
+    ssh_channel.request_exec('cat')
+    ssh_channel.send_eof()
+
+    rc = -1
+    while rc == -1:
+        ssh_channel.poll(timeout=5)
+        rc = ssh_channel.get_channel_exit_status()
+    assert rc == 0
+
+
+def test_send_signal(ssh_channel):
+    """Test send_signal correctly forwards signal to the process."""
+    ssh_channel.request_exec('bash -c \'trap "exit 1" SIGUSR1; echo ready; sleep 5; exit 0\'')
+
+    # Wait until the process is ready to receive signal
+    output = ''
+    while not output.startswith('ready'):
+        ssh_channel.poll(timeout=1000)
+        output += ssh_channel.recv().decode('utf-8')
+
+    # Send SIGINT
+    ssh_channel.send_signal('USR1')
+
+    rc = -1
+    while rc == -1:
+        ssh_channel.poll(timeout=5)
+        rc = ssh_channel.get_channel_exit_status()
+
+    assert rc == 1
+
+
+def test_recv_eof(ssh_channel):
+    """
+    Test that reading EOF does not raise error.
+
+    SystemError: Negative size passed to PyBytes_FromStringAndSize
+    """
+    ssh_channel.request_exec('exit 0')
+    ssh_channel.poll(timeout=1000)
+    ssh_channel.recv()
+
+
+def test_is_eof(ssh_channel):
+    """Test that EOF-state is correctly obtained with is_eof."""
+    ssh_channel.request_exec('exit 0')
+    ssh_channel.poll(timeout=5000)  # noqa: WPS432
+    assert ssh_channel.is_eof()
+
+
+def test_destructor(sshd_addr, ssh_clientkey_path):
+    """
+    Garbage collector can destroy session before channel.
+
+    Test that this event does not cause a segfault in channels destructor.
+    """
+    def _do_not_crash():  # noqa: WPS430
+        ssh_session = Session()
+        ensure_ssh_session_connected(ssh_session, sshd_addr, ssh_clientkey_path)
+        ssh_channel = ssh_session.new_channel()  # noqa: F841
```

Comment:
Would it be easier to trigger if this is linked to an external object?
```suggestion
        _do_not_crash.ssh_channel = ssh_session.new_channel()  # noqa: F841  # setting a non-accessed var is needed for testing GC
```

Reply:
I don't understand what you mean, can you ellaborate? But it is very easy to trigger it. Really just open session and channel and then let the program exit.

Reply:
@pbrezina but in tests you have a separate nested function in hopes the variables in its scope would be garbage-collected in a breaking matter, right?

Reply:
Right. It needs to be garbage collected, so it needs to go out of scope.